docs/loadtesting.md

```diff
@@ -0,0 +1,42 @@
+# Load Testing
+
+We want to observe how the server behaves under realistic load. To
+keep things realistic, our approach should:
+
+- use real HTTPS requests against a load-testing system configured similarly to the real system
+- use requests that look as much as possible like the real thing, e.g. with expected cookies
+- figure out how to load test Arlo without also load-testing Auth0 for authentication
+- whatever hacks we put in for authentication (it should be clear by
+  now we'll need some), they should not affect production code.
+- scale up well with ideally minimal infrastructural investment so
+  we're not manually spinning up dozens of machines ourselves.
+
+## Authentication Hack for Load Testing
+
+We have superadmin features that allow for impersonation of any user,
+and which uses the same cookies as normal logins. Let's reuse that.
+
+We'll build a superadmin bypass using a special HTTP header. Because
+we don't ever want that code in production, we'll keep that code in a
+special branch, `loadtesting`, that we'll need to rebase every so
+often on `master`. We'll also require a special environment variable
+be set as an extra defense-in-depth measure, just in case we
+mistakenly merge that branch into `master`. Belt _and_ suspenders.
+
+## Tool
+
+One tool that some of us have used with success is
+[Artillery](https://artillery.io), and in particular [Serverless
+Artillery](https://github.com/Nordstrom/serverless-artillery), which
+combines the same test scripting language with AWS Lambda deployment
+for instant scaling without overhead of setting up machines.
```

Comment:
will we be able to reuse these as integration tests locally/on CI? or, put another way, can we use integration tests as the core script for load testing?

Reply:
I don't think load testing makes sense in integration testing. It would be very expensive and time-taking.

Reply:
Ooh, sorry, I wasn't clear.

I just meant that the test scripts you're writing for load testing will be walking through the user flows in the exact same ways we want to walk through them in integration tests. So it would be nice if we could reuse those test scripts (but for integration tests, only running the flow once)

Reply:
Aha I see. So that's an interesting conundrum, because that changes the requirements in a way that might increase risk. Let's discuss.